internal/daemon/daemon.go

```diff
@@ -299,13 +299,21 @@ func logit(handler http.Handler) http.Handler {
 		t0 := time.Now()
 		handler.ServeHTTP(ww, r)
 		t := time.Now().Sub(t0)
-		if !strings.Contains(r.URL.String(), "/v1/changes/") {
-			if strings.HasSuffix(r.RemoteAddr, ";") {
-				logger.Debugf("%s %s %s %s %d", r.RemoteAddr, r.Method, r.URL, t, ww.s)
-				logger.Noticef("%s %s %s %d", r.Method, r.URL, t, ww.s)
-			} else {
-				logger.Noticef("%s %s %s %s %d", r.RemoteAddr, r.Method, r.URL, t, ww.s)
-			}
+
+		if r.Method == "GET" && (strings.HasPrefix(r.URL.Path, "/v1/changes/") || r.URL.Path == "/v1/system-info") {
+			// Don't log GET /v1/changes/{change-id} as that's polled quickly
+			// by clients when waiting for a change (e.g., service starting).
+			// Also don't log GET /v1/system-info to avoid it filling logs with
+			// noise when that endpoint is used as a kind of health check
+			// (Juju hits it every 5s, for example).
+			return
```

Comment:
Thanks for the nice explanation.

Can we please just return the logic within the branch instead of having the early return?  I get the pattern and use it often as well, but it seems wrong for this specific case. This function is an the main API handler, but the conditional is basically saying "Should I log?", and not "Should I go on with this request?". It will be very weird to add something at the end of this function and find out it doesn't work because something shouldn't be logged.

Reply:
Fair - done.